### PR TITLE
Update default HPA API Version to `v2` and add support for behavior

### PR DESCRIPTION
**Pull request by grieshaber**
### What does this PR do?

This MR updates the API-Version of the hpa to autoscaling/v2beta2. Since k8s 1.18, this api-version adds support for custom scaling behavior. Therefore i added some example values and the corresponding property in hpa.yaml.

### Motivation
Requested by @Xyaren in #515

### More

- [X] Yes, I updated the [chart version](https://github.com/traefik/traefik-helm-chart/blob/9b32ed1b414cc0be1ad46bcb335fcfc93ded06f3/traefik/Chart.yaml#L5)

### Additional Notes

I wrapped the behavior property in the hpa in an additional if clause, because it is only supported in clusters >= 1.18, so it won't break things and needs to be enabled explicitly.

### CODE REVIEW

**Comment by junaid-ali** [comment]
@kevinpollet any update on this one, could you please have a look at this again?

**Comment by cilindrox** [comment]
this is also relevant since v2beta2 allows specifying multiple metrics for the HPA to scale on. Right now we're constrained to a single metric.

**Comment by grieshaber** [comment]
I rebased my branch to the latest changes and would appreciate someone having a look at this PR.

File: traefik/templates/hpa.yaml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {{- if .Values.autoscaling.enabled }}
-apiVersion: autoscaling/v2beta1
+apiVersion: autoscaling/v2beta2
```
Comment (jburke-godaddy) [inline traefik/templates/hpa.yaml:2]:
```suggestion
{{- if .Capabilities.APIVersions.Has "autoscaling/v2beta2/HorizontalPodAutoscaler" }}
apiVersion: autoscaling/v2beta2
{{- else if .Capabilities.APIVersions.Has "autoscaling/v2beta1/HorizontalPodAutoscaler" }}
apiVersion: autoscaling/v2beta1
{{- else }}
{{- fail "\n\n ERROR: You must have at least autoscaling/v2beta1 to use HorizontalPodAutoscaler" }}
{{- end }}
```
Similar adaption is in: https://github.com/traefik/traefik-helm-chart/blob/859adf66447d8242ae6c697fc901473d4684c282/traefik/templates/ingressclass.yaml#L2-L10

This code should allow the template to adapt to older versions.

**Review by jburke-godaddy — CHANGES_REQUESTED** [review]

**Comment by mrx88** [comment]
FYI - maybe due to helm version problems (did not verify),  but this part is not rendering correctly when using the example configuration with Helm v3.10.1:
```
{{- if .Values.autoscaling.behavior }}
  behaviour: 
{{ toYaml .Values.autoscaling.behavior | indent 4 }}
{{- end }}
```
Output is:
```
 behavior:
    scaleDown:
      policies:
      - periodSeconds: 60
        type: Pods
        value: 1
      stabilizationWindowSeconds: 300
```

but should be:
```
  behavior:
    scaleDown:
      stabilizationWindowSeconds: 300
      policies:
      - type: Pods
        value: 1
        periodSeconds: 60
```
(Note the -type is missing).

**Comment by mloiseleur** [comment]
Hello @mrx88,

There is `type: Pods` in the output. It's expected (and alphabetically ordered) output. 
In this example, `periodSeconds`, `type` and `value` are part of the same policy. It doesn't matter if `periodSeconds` is before or after `type` as long as they are in the same policy.

**Comment by mrx88** [comment]
> Hello @mrx88,
> 
> There is `type: Pods` in the output. It's expected (and alphabetically ordered) output. In this example, `periodSeconds`, `type` and `value` are part of the same policy. It doesn't matter if `periodSeconds` is before or after `type` as long as they are in the same policy.

I was pointing out to "-" char missing for Type, but if it works that way too then no worries ;)

**Pull request merged**